File: curso_java/10_clases_objetos/ejemplos/Rectangulo.java
/*
 * Una clase con atributos y metodos de instancia, mas un main para probarla.
 */
public class Rectangulo {
    double base;      // atributos: cada objeto tiene los suyos
    double altura;

    double area() {                 // metodo de instancia
        return base * altura;
    }
    double perimetro() {
        return 2 * (base + altura);
    }

    public static void main(String[] args) {
        Rectangulo r = new Rectangulo();   // crear un objeto
        r.base = 4;
        r.altura = 3;
        System.out.println("Area: " + r.area());            // 12
        System.out.println("Perimetro: " + r.perimetro());  // 14

        Rectangulo otro = new Rectangulo();
        otro.base = 10;
        otro.altura = 2;
        System.out.println("Area del otro: " + otro.area()); // 20 (objeto independiente)
    }
}

File: curso_c_cpp/cpp/04_clases_objetos/ejemplos/rectangulo.cpp
/*
 * Primera clase: Rectangulo, con atributos publicos y un metodo.
 */
#include <iostream>
using namespace std;

class Rectangulo {
public:
    double base;
    double altura;

    double area() {
        return base * altura;
    }
    double perimetro() {
        return 2 * (base + altura);
    }
};

int main() {
    Rectangulo r;              // objeto
    r.base = 4;
    r.altura = 3;
    cout << "Area: " << r.area() << endl;            // 12
    cout << "Perimetro: " << r.perimetro() << endl;  // 14
    return 0;
}
